daemon/create.go

```diff
@@ -152,5 +153,10 @@ func (daemon *Daemon) VolumeCreate(ctx context.Context, name, driverName string,
 	if err != nil {
 		return nil, err
 	}
+
+	// keep "docker run -v existing_volume:/foo --volume-driver other_driver" work
+	if (driverName != "" && v.DriverName() != driverName) || (driverName == "" && v.DriverName() != volume.DefaultDriverName) {
```

Comment:
I wonder if we should move this logic to the before calling `daemon.volumes.Create` in this same function. I'm worried we'd end up sending unnecessary creation requests to the drivers. What about something like:

```go
v, err := daemon.volumes.Get(name)
if err != errors.ErrorNoSuchVolume || v.DriverName() != driverName {
   // Return error here
}

v, err := daemon.volumes.Create...
```